pkg/sql/sqlbase/kvfetcher.go

```diff
@@ -357,48 +353,29 @@ func (f *txnKVFetcher) fetch(ctx context.Context) error {
 	return nil
 }
 
-func (f *txnKVFetcher) batchIsLimited(batchSize int64) bool {
-	if f.useBatchLimit {
-		// f.batchIdx - 1 refers to the most recent fetch.
-		return batchSize == f.getBatchSizeForIdx(f.batchIdx-1)
-	}
-	return false
```

Comment:
return f.useBatchLimit && batchSize == f.getBatchSizeForIdx(f.batchIdx-1)

Reply:
But I'm removing this code, not adding it! :)